src/test/java/com/icthh/xm/uaa/security/DomainTokenServicesUnitTest.java

```diff
@@ -155,6 +180,40 @@ public void testRefreshToken() throws Exception {
         assertEquals(USER_KEY, ((Map<String, String>) auth.getDetails()).get(AUTH_USER_KEY));
     }
 
+    @Test
+    public void testRefreshTokenForDeactivatedUser() throws Exception {
+
+        when(userService.findOneByLogin(LOGIN)).thenAnswer(invocation -> {
+            User user = new User();
+            user.setActivated(false); // user is not active
+            user.setUserKey(USER_KEY);
+            return Optional.of(user);
+        });
+
+        OAuth2AccessToken accessToken = tokenServices.createAccessToken(createAuthentication(LOGIN, TENANT, ROLE));
+
+        assertNotNull(accessToken);
+        assertNotNull(accessToken.getRefreshToken());
+        assertNotNull(accessToken.getRefreshToken().getValue());
+
+        Map<String, String> params = new HashMap<>();
+        params.put("grant_type", "refresh_token");
+        params.put("refresh_token", accessToken.getRefreshToken().getValue());
+
+        TokenRequest tokenRequest = new TokenRequest(params, CLIENT, null, "refresh_token");
+        try {
+            tokenServices.refreshAccessToken(accessToken.getRefreshToken().getValue(), tokenRequest);
+            fail( "Expected UserDeniedAuthorizationException to be thrown" );
+        } catch (UserDeniedAuthorizationException e){
+            // ignore
+        }
+
+        verify(authenticationManager, never()).authenticate(any());
+        verify(tokenStore, never()).removeAccessTokenUsingRefreshToken(any());
+        verify(tokenStore, never()).storeAccessToken(eq(accessToken), any());
+        verify(authenticationRefreshProvider, never()).refresh(any());
```

Comment:
better to user no interactions with mock noMoreIntercationWithMock() or smth like this

Reply:
+